src/payment/data/sagas.test.js

```diff
@@ -162,6 +239,39 @@ describe('saga tests', () => {
       );
     });
 
+    it('should update basket data without calling discount check API', async () => {
```

Comment:
Added this test to prove we didn't call the discount endpoint if we have no products.